### PR TITLE
Moving PooledTxs requesting to PooledTxsRequestor

### PR DESCRIPTION
**Pull request by marcindsobczak**

### CODE REVIEW

**Comment by tkstanczak** [comment]
https://dictionarykiwi.com/2016/10/17/whats-the-difference-between-requester-and-requestor/

Use Requestor in technical contexts :) (this comment is sooo meaningful...)

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/PooledTxsRequester.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            IList<Keccak> discoveredTxHashes = FilterHashes(hashes);
+            
+            if (discoveredTxHashes.Count != 0)
+            {
+                send(new GetPooledTransactionsMessage(discoveredTxHashes.ToArray()));
+                Metrics.Eth65GetPooledTransactionsRequested++;
+            }
+        }
+
+        public IList<Keccak> FilterHashes(IList<Keccak> hashes)
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/PooledTxsRequester.cs:49]: What is the idea behind this method?

Comment (marcindsobczak) [reply src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/PooledTxsRequester.cs]: This method has all of received newPooledTxHashes as an argument and is checking if each of these hashes is in hachCache or in pendingHashes - if not present in any, then adds to pendingHashes and to the list of discovered hashes, which is returned back to RequestTransactions, where this hashes are used to request PooledTransactions. I made FilterHashes as separate method for the testing reasons - most of the class logic is here and with method in this shape it's easy to test pendingHashes behavior

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/IPooledTxsRequestor.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+using System;
+using System.Collections.Generic;
+using Nethermind.Core.Crypto;
+
+namespace Nethermind.Network.P2P.Subprotocols.Eth.V65
+{
+    public interface IPooledTxsRequestor 
+    {
+        void RequestTransactions(Action<GetPooledTransactionsMessage> send, IList<Keccak> hashes);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/IPooledTxsRequestor.cs:26]: IReadOnlyList?

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/PooledTxsRequestor.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            IList<Keccak> discoveredTxHashes = FilterHashes(hashes);
+            
+            if (discoveredTxHashes.Count != 0)
+            {
+                send(new GetPooledTransactionsMessage(discoveredTxHashes.ToArray()));
+                Metrics.Eth65GetPooledTransactionsRequested++;
+            }
+        }
+
+        public IList<Keccak> FilterHashes(IList<Keccak> hashes)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/PooledTxsRequestor.cs:49]: IReadOnlyList?
public?
naming -> GetAndMarkUnknownHashes?

**Review by LukaszRozmej — APPROVED** [review]
Good, just minor details

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/PooledTxsRequestor.cs (hunk 3 above)
Comment (marcindsobczak) [reply src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/PooledTxsRequestor.cs]: I need it public to be able to test it separately from RequestTransactions

File: src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/PooledTxsRequestor.cs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            IList<Keccak> discoveredTxHashes = GetAndMarkUnknownHashes(hashes);
+            
+            if (discoveredTxHashes.Count != 0)
+            {
+                send(new GetPooledTransactionsMessage(discoveredTxHashes.ToArray()));
+                Metrics.Eth65GetPooledTransactionsRequested++;
+            }
+        }
+
+        public IList<Keccak> GetAndMarkUnknownHashes(IReadOnlyList<Keccak> hashes)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/PooledTxsRequestor.cs:49]: should be private

**Pull request merged**